subsys/net/ip/net_if.c

```diff
@@ -3639,6 +3801,154 @@ int net_if_down(struct net_if *iface)
 	return 0;
 }
 
+static int lookup_ifaddr(struct net_if *iface, struct sockaddr *addr,
+			 struct net_if_addr **ifaddr)
+{
+	if (iface == NULL || addr == NULL) {
+		return -EINVAL;
+	}
+
+	if (IS_ENABLED(CONFIG_NET_IPV6) && addr->sa_family == AF_INET6) {
+		*ifaddr = net_if_ipv6_addr_lookup_by_iface(iface,
+						&net_sin6(addr)->sin6_addr);
+		if (*ifaddr == NULL &&
+		    net_ipv6_is_addr_unspecified(&net_sin6(addr)->sin6_addr)) {
+			return -EADDRNOTAVAIL;
+		}
+
```

Comment:
nit: Extra empty line.

Reply:
Actually this extra NL and the other one you mentioned, I added them on purpose so that the lines would not look so cluttered.